### ISSUE TITLE
Problems deploying ASP.NET 5 RC1 to IIS 7.5

### ISSUE THREAD

**Issue by jcasale** [aspnet/Home#1188]
I have a simple MVC app for which I am publishing to the filesystem from VS and attempting to deploy to the default IIS virtual server. I have followed http://docs.asp.net/en/latest/publishing/iis.html and installed 64 bit HTTP Platform Handler 1.2, unlocked the system.webServer/handlers configuration and created a new app pool as per the doc.

If I run web.cmd, the site runs fine when accessed from the local host. If I access it through IIS, the home page loads, but when accessing any other controller, the default controller's error page is served. The workaround at https://github.com/aspnet/Hosting/issues/416 is not applicable to me as far as I know, however I tried it passing an empty string as the first parameter without luck.

I am new to asp.net and not clear no how to gather any additional info that might help. Any ideas would be greatly appreciated.

**Comment by GuardRex** [comment]
@jcasale If by "home page loads" you mean an Mvc view loads, I would guess this is a problem with your routing.

**Comment by jcasale** [comment]
@GuardRex I have a single route "{controller=Home}/{action=Index}" and 3 controllers, {Home,Project,Client}Controllers, so the HomeController does function and load the Index view. The other two controllers run when just web.cmd or IISExpress in Visual Studio serves them.

Is there a way to debug the routing engine to see what its receiving and attempting to do?

Thanks for the assistance.

**Comment by GuardRex** [comment]
@jcasale Good. If you're getting even just one MVC view, then it's probably just going to be routing-related. One of the things under heavy development right now are the MVC docs, which will help in the future (I'm looking forward to reading them myself) ... http://aspnetmvc.readthedocs.org/projects/mvc/en/latest/. Unfortunately, the controllers docs aren't there yet.

In terms of figuring out routing, there are some blogs and posts here and there (Google "MVC6 routing" and/or "MVC6 attribute routing"), but you might just need to look at a good sample, and the "everything including the kitchen sink" sample is the [Music Store sample](https://github.com/aspnet/MusicStore/tree/dev/src/MusicStore). However, I don't think that has much attribute routing in it (my favorite way to go). You can see in the Music Store sample how they have the controllers setup there and see what deltas they have compared to what you have.

There is also good info in ...

http://aspnetmvc.readthedocs.org/projects/mvc/en/latest/migration/migratingfromwebapi2.html
http://stephenwalther.com/archive/2015/02/07/asp-net-5-deep-dive-routing
http://www.codeproject.com/Articles/1035231/MVC-attribute-routing-new-features-and-differences
http://blogs.msdn.com/b/webdev/archive/2014/06/20/asp-net-vnext-routing-overview.aspx

**Comment by jcasale** [comment]
After reading labbing up samples I couldn't see how it was routing related, so I rolled out a new dev virtual machine with Visual Studio 2015 Update 1 (previous was not update 1), installed AspNet5.ENU.RC1_Update1.exe (previous was only RC1) and setup a new template ASP.NET 5 Web Application and started adding in class by class and call by call from the existing project.

Turns out it was uncaught exception in the first non Home controllers Index view creation. A call to DirectoryInfo.EnumerateDirectories() threw an UnauthorizedAccessException which _isn't_ listed in the calls meta data:(

This user error on my part could have been caught more easily if I could attach visual studio to the website remotely and step through it, but I can not find docs on this for ASP.NET 5. Is it possible?

Thanks for the patience and the info above.
